app/controllers/questionnaires_controller.rb

```diff
@@ -572,20 +597,27 @@ def question_params
 
   def export
     @questionnaire = Questionnaire.find(params[:id])
-
-    csv_data = QuestionnaireHelper.create_questionnaire_csv @questionnaire, session[:user].name
-
-    send_data csv_data,
-              type: 'text/csv; charset=iso-8859-1; header=present',
-              disposition: "attachment; filename=questionnaires.csv"
+    questionnaire_id = params[:id] unless params[:id].nil?
+    questions = Question.where("questionnaire_id = " + questionnaire_id.to_s).sort { |a,b| a.seq <=> b.seq }
+    respond_to do |format|
+      format.csv { send_data @questionnaire.to_csv(questions) }
+    end
+    redirect_to Questionnaire,notice: $ERROR_INFO and return if @questionnaire.nil?
   end
 
   def import
-    @questionnaire = Questionnaire.find(params[:id])
-
-    file = params['csv']
-
-    @questionnaire.questions << QuestionnaireHelper.get_questions_from_csv(@questionnaire, file)
+    questionnaire_id = (params[:id])
+    begin
+      uploaded_io = params[:csv]
+      File.open(Rails.root.join('spec', 'features/import_export_csv_oss/', uploaded_io.original_filename), 'wb') do |file|
+        file.write(uploaded_io.read)
+      end
+        file_data = File.read(Rails.root.join('spec/features/import_export_csv_oss/'+uploaded_io.original_filename))
+        a = QuestionnaireHelper.get_questions_from_csv(file_data,params[:id])
```

Comment:
We were planning to use this in modifying the error message. Can we push the changes after removing this?